source/read/change-streams.txt

```diff
@@ -0,0 +1,261 @@
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: watch, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use a **change stream** to monitor real-time
+changes to your data. A change stream is a {+mdb-server+} feature that
+allows your application to subscribe to data changes on a collection, database,
+or deployment.
+
+When using the {+php-library+}, you can create an instance of ``MongoDB\ChangeStream`` to
+monitor data changes, such as updates, insertions, and deletions.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your PHP application, instantiate a ``MongoDB\Client`` that connects to an Atlas cluster
+and assign the following value to your ``collection`` variable:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :language: php
+   :dedent:
+   :start-after: start-db-coll
+   :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Open a Change Stream
+--------------------
+
+To open a change stream, call the ``watch()`` method. The instance on which you
+call the ``watch()`` method determines the scope of events that the change
+stream monitors. You can call the ``watch()`` method on instances of the following
+classes:
+
+- ``MongoDB\Client``: Monitor all changes in the MongoDB deployment
+- ``MongoDB\Database``: Monitor changes in all collections in the database
+- ``MongoDB\Collection``: Monitor changes in the collection
+
+The following example opens a change stream on the ``restaurants`` collection
+and outputs changes as they occur:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :start-after: start-open-change-stream
+   :end-before: end-open-change-stream
+   :language: php
+   :dedent:
+
+To begin watching for changes, run the preceding code. Then, in a separate
+application or shell, modify the ``restaurants`` collection. The following
+example updates a document that has a ``name`` field value of ``'Blarney Castle'``:
+
+.. _php-change-stream-update:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :start-after: start-update-for-change-stream
+   :end-before: end-update-for-change-stream
+   :language: php
+   :dedent:
+
+When you update the collection, the change stream application prints the change
+as it occurs. The printed change event resembles the following output:
+
+.. code-block:: none
+   :copyable: false
+
+   {"_id":{"_data":"..."},"operationType":"update","clusterTime":{"$timestamp":{...},
+   "wallTime":{"$date":...},"ns":{"db":"sample_restaurants","coll":"restaurants"},
+   "documentKey":{"_id":{"$oid":"..."}},"updateDescription":{"updatedFields":
+   {"cuisine":"Irish"},"removedFields":[],"truncatedArrays":[]}}}
```

Comment:
If `json_encode()` is used in the PHP example, I don't think you'll see keys like "$timestamp", "$date", and "$oid" in the output here. The BSON API will need to be used to ensure Extended JSON syntax is produced.